.github/workflows/pipeline.yml

```
@@ -1,136 +1,220 @@
-name: Build and lint
-
+name: Build, lint and typecheck examples
+ 
 on:
     # Triggers the workflow on push or pull request events but only for the main branch
     push:
         branches: [main, release**]
     pull_request:
         branches: [main, release**]
-
+        # Don't run on draft PR's, see: https://github.com/orgs/community/discussions/25722#discussioncomment-3248917
+        types: [opened, synchronize, reopened, ready_for_review]
     # Allows us to run the workflow manually from the Actions tab
     workflow_dispatch:
 
 env:
+  NODE_VERSION: 18.16.0
+  RUST_VERSION: 1.62
```

Seems we use this in other projects:
```suggestion
  RUST_VERSION: 1.68
```